File: server/get_vc_funding_data.py
# lambda endpoint for our frontend to get the vc funding data
import os
from dotenv import load_dotenv
import logging
from pymongo import MongoClient

load_dotenv()
logging.basicConfig(level=logging.getLevelName(os.getenv('LOGGING_LEVEL')),
                    format="[%(levelname)s | %(asctime)s | %(filename)s:%(lineno)s] : %(message)s",
                    datefmt='%Y-%m-%d %H:%M:%S')

MONGO_CONNECTION_STR = os.getenv('DB_CONNECTION_STR')
DB_NAME = os.getenv('DB_NAME')
DB_FUNDING_COLLECTION = os.getenv('DB_FUNDING_COLLECTION')
NUM_FUNDING_ENTRIES = os.getenv('NUM_FUNDING_ENTRIES')
DB_EXPIRY_DATE_COLLECTION = os.getenv('DB_EXPIRY_DATE_COLLECTION')


def get_funding_data():
    client = MongoClient(MONGO_CONNECTION_STR)
    entries = {'articles': [], 'expiry_date': None}
    try:
        db = client[DB_NAME]
        collection = db[DB_FUNDING_COLLECTION]
        logging.info("Successfully connected to db")
        collection.create_index('date')
        data = list(collection.find({"company_name": {"$ne": None}},{"_id":0}).sort({ "date": -1}).limit(int(NUM_FUNDING_ENTRIES)))
        collection = db[DB_EXPIRY_DATE_COLLECTION]
        expiry_date = collection.find_one({"title": "expiry_date"}, {'_id': 0})
        entries = {'articles': data, 'expiry_date': expiry_date['expiry_date']}

    except Exception as e:
        logging.error(f"Error while getting data: {e}")
    finally:
        client.close()
        logging.info("Returning info to FASTAPI")
        return entries


if __name__ == '__main__':
    logging.info(len(get_funding_data()))

File: server/getVcFundingData.js
import express, { Router } from "express";
import serverless from "serverless-http";
import { MongoClient } from "mongodb";
import dotenv from "dotenv";

dotenv.config();
const client = new MongoClient(process.env.DB_CONNECTION_STR);
let conn;
try {
  conn = await client.connect();
} catch(e) {
  console.error(e);
}
let db = conn.db(process.env.DB_NAME);


const api = express();
const router = Router();
api.use("/.netlify/functions/server", router);

router.get("/vc_funding_data", async (req, res) =>
{
    let collection = await db.collection(process.env.DB_FUNDING_COLLECTION);
    let results = await collection.find({"company_name": {"$ne": null}},{"_id":0})
    .sort({ "date": -1})
    .limit(parseInt(process.env.NUM_FUNDING_ENTRIES) || 1500)
    .toArray();
  res.send(results).status(200);
} );



// const port = 8000
// api.listen(port, () => {
//     console.log(`Example app listening on port ${port}`)
//   })

module.exports = api
module.exports.handler = serverless(api)